starter-code/src/clue.js

```diff
@@ -84,11 +84,206 @@ name: Guest House
+    age: 31,
+    image: "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+    occupation: "Actor"
+};
+
+var mrsPeacock = {
+    userName: "mrsPeacock",
+    firstName: "Eleanor",
+    lastName: "Peacock",
+    color: "blue",
+    description: "She is from a wealthy family and uses her status and money to earn popularity",
+    age: 36,
+    image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+    occupation: "Socialité"
+};
+
+var mrMustard = {
+    userName: "mrMustard",
+    firstName: "Jack",
+    lastName: "Mustard",
+    color: "yellow",
+    description: "He is a former football player who tries to get by on his former glory",
+    age: 62,
+    image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+    occupation: "Retired Football player"
+};
+
+var rope = {
+    name: "rope",
+    weight: 10
+};
+
+var knife = {
+    name: "knife",
+    weight: 8
+};
+
+var candlestick = {
+    name: "candlestick",
+    weight: 2
+};
+
+var dumbbell = {
+    name: "dumbbell",
+    weight: 30
+};
+
+var poison = {
+    name: "poison",
+    weight: 2
+};
+
+var axe = {
+    name: "axe",
+    weight: 15
+};
+
+var bat = {
+    name: "bat",
+    weight: 13
+};
+
+var trophy = {
+    name: "trophy",
+    weight: 25
+};
+
+var pistol = {
+    name: "pistol",
+    weight: 2
+};
+
+var dinningRoom = {
+    name: "Dinning Room"
+};
+
+var conservatory = {
+    name: "Conservatory"
+};
+
+var kitchen = {
+    name: "Kitchen"
+};
+
+var study = {
+    name: "Study"
+};
+
+var library = {
+    name: "Library"
+};
+
+var billiardRoom = {
+    name: "Billiard Room"
+};
+
+var lounge = {
+    name: "Lounge"
+};
+
+var ballroom = {
+    name: "Ballroomm"
+};
+
+var hall = {
+    name: "Hall"
+};
+
+var spa = {
+    name: "Spa"
+};
+
+var livingRoom = {
+    name: "lLiving Room"
+};
+
+var observatory = {
+    name: "Observatory"
+};
+
+var theater = {
+    name: "Theater"
+};
+
+var guestHouse = {
+    name: "Guest House"
+};
+
+var patio = {
+    name: "Patio"
+};
+
+
+// Characters Collection---------------------------------------
+var charactersArray = [
+    mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard
+];
+//randomSelector(charactersArray);
+
+
+// Rooms' Collection-----------------------------------------
+var roomsArray = [
+    dinningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio
+];
+//randomSelector(roomsArray);
+
+
+// Weapons Collection-------------------------------------------
+var weaponsArray = [
+    rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol
+];
+//randomSelector(weaponsArray);
+
+
+// Functions-------------------------------------------------------
+function randomSelector(cardsArray) {
+    return cardsArray[Math.floor(Math.random() * cardsArray.length)];
+}
+
+function pickMistery(picked) {
```

Comment:
here, you don't need to pass anything to the function. 

For this function to work you need to call your randomSelector() here with the name of an array (eg weaponArray) passed to it. Once you have picked a random item from the array, you need to push that to an array that you'll use later to reveal the mystery 

for example - 

``` 
function pickMistery() {
  var resultArray = [];
  var killer = randomSelector(charactersArray);

  resultArray.push(killer);

}
```